cloudwatch_logs_common/include/cloudwatch_logs_common/utils/cloudwatch_facade.h

```diff
@@ -115,12 +123,13 @@ class CloudWatchFacade
 
 protected:
   CloudWatchFacade() = default;
+  
+  std::shared_ptr<Aws::CloudWatchLogs::CloudWatchLogsClient> cw_client_;
```

Comment:
Why a shared pointer? Unique pointer seems more suitable as that value is only used by this class in the production code. You can access the raw pointer during test with the `get` method in order to do assertions

Reply:
+1